internal/transport/transport_test.go

```diff
@@ -2746,3 +2756,85 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		t.Errorf("Context timed out")
 	}
 }
+
+// hangingConn is a net.Conn wrapper for testing, simulating hanging connections
+// after a GOAWAY frame is sent, of which Write operations pause until explicitly signaled
+// or a timeout occurs.
+type hangingConn struct {
+	net.Conn
+}
+
+func (hc *hangingConn) Read(b []byte) (n int, err error) {
+	n, err = hc.Conn.Read(b)
+	return n, err
+}
+
+func (hc *hangingConn) Write(b []byte) (n int, err error) {
+	n, err = hc.Conn.Write(b)
+	if isGreetingsDone.Load() == true {
+		// Add a delay which is more than goAwayLoopyWriterTimeout
+		time.Sleep(2 * time.Second)
```

Comment:
Please don't use a sleep here.  Instead use a channel or `grpcsync.Event` that comes from the dialer.  We can trigger it after the test case passes.

Reply:
Updated